### ISSUE TITLE
shairport-sync.spec and libpopt (broken)

### ISSUE THREAD

**Issue by martinoneutrino** [mikebrady/shairport-sync#347]
In commit  90799d45687fe6df5911008b7013af4da7c77887:

> Requires: popt was changed to Requires: libpopt-dev

This breaks an rpm built on Fedora, as there is nothing that provides "libpopt-dev". Removing the line completely allows the rpm installed (the dependency on libpopt.so.0 is still in the built rpm).

Also, I would think that "libpopt-dev" would be the dev build dependency, and shouldn't be in Requires for Debian either.

**Comment by mikebrady** [comment]
Thanks for this – I'd defer to @vru1 on this one.

**Comment by vru1** [comment]
Well, I can only say that `popt` did not work on Debian Jessie, it was reported as an unknown dependency while compiling a .deb package using `checkinstall.` After I've changed `popt` to `libpopt-dev`, the package was created without any error. Here is a list of Debian packages related to `popt`:
https://packages.debian.org/search?keywords=popt

**Comment by martinoneutrino** [comment]
Does it work without `libpopt-dev`? I wouldn't think that a development library package should be a prerequisite (https://packages.debian.org/search?keywords=libpopt-dev).

**Comment by vru1** [comment]
Now I've tried to remove `Requires:       libpopt-dev` from `shairport-sync.spec` and the .deb package was created without any error message.
I've also tried to remove the whole section 
```
Requires:       libpopt-dev
Requires:       openssl
Requires:       avahi
Requires:       libdaemon
Requires:       alsa-lib
Requires:       soxr
```
from `shairport-sync.spec` and the package still could be created. I can't say if this is going to have some consequences elsewhere nor what the purpose of that section is, I don't have enough experience. Couldn't find anything here:
https://www.debian.org/doc/debian-policy/ch-relationships.html

**Comment by martinoneutrino** [comment]
That section indicates each library that is required to be available at install (run) time. If you removed those libraries from your machine and then installed (without the dependencies listed), you'd get an error when starting the program.
The rpmbuild process is pretty smart (on Fedora at least) and actually includes most of the required libraries already (not sure about Debian). You can see below that `libpopt` is already required:

> $ rpm -q --requires shairport-sync
> /bin/sh
> alas-lib
> avahi
> config(shairport-sync) = 2.8.4.4-1
> libasound.so.2()(64bit)
> libasound.so.2(ALSA_0.9)(64bit)
> libasound.so.2(ALSA_0.9.0rc4)(64bit)
> libavahi-client.so.3()(64bit)
> libavahi-common.so.3()(64bit)
> libc.so.6()(64bit)
> libc.so.6(GLIBC_2.14)(64bit)
> libc.so.6(GLIBC_2.15)(64bit)
> libc.so.6(GLIBC_2.2.5)(64bit)
> libc.so.6(GLIBC_2.3)(64bit)
> libc.so.6(GLIBC_2.3.4)(64bit)
> libc.so.6(GLIBC_2.4)(64bit)
> libconfig.so.9()(64bit)
> libcrypto.so.10()(64bit)
> libcrypto.so.10(libcrypto.so.10)(64bit)
> libdaemon
> libdaemon.so.0()(64bit)
> libm.so.6()(64bit)
> libm.so.6(GLIBC_2.2.5)(64bit)
> **libpopt.so.0()(64bit)**
> **libpopt.so.0(LIBPOPT_0)(64bit)**
> libpthread.so.0()(64bit)
> libpthread.so.0(GLIBC_2.2.5)(64bit)
> libpthread.so.0(GLIBC_2.3.2)(64bit)
> libpthread.so.0(GLIBC_2.3.3)(64bit)
> librt.so.1()(64bit)
> librt.so.1(GLIBC_2.2.5)(64bit)
> libsoxr.so.0()(64bit)
> libssl.so.10()(64bit)
> openssl
> rpmlib(CompressedFileNames) <= 3.0.4-1
> rpmlib(FileDigests) <= 4.6.0-1
> rpmlib(PayloadFilesHavePrefix) <= 4.0-1
> rpmlib(PayloadIsXz) <= 5.2-1
> rtld(GNU_HASH)
> soxr
> 

Similarly, the entries for `openssl, avahi, libdaemon, alsa-lib, soxr` are all redundant, and I think should be removed.

**Comment by mikebrady** [comment]
Thanks for this contribution. Please open a new issue if necessary.
